server/src/main/java/org/tctalent/server/service/db/aws/S3ResourceHelper.java

```diff
@@ -34,292 +51,285 @@
+      InputStream is = new ByteArrayInputStream(bytes);
+      uploadFile(bucket, is, bytes.length, key, contentType);
+    } catch (FileUploadException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new FileUploadException("Error uploading string content to: " + key, e);
     }
-
-    public boolean doesBucketExist(String bucket) {
-        return amazonS3.doesBucketExist(s3Bucket + "bucket");
+  }
+
+  public void uploadFile(String bucket,
+      InputStream content,
+      long contentLength,
+      String key,
+      String contentType)
+      throws FileUploadException {
+    try {
+      ObjectMetadata objectMetadata = new ObjectMetadata();
+      objectMetadata.setContentLength(contentLength);
+      objectMetadata.setContentType(contentType);
+      Upload upload = transferManager.upload(bucket, key, content, objectMetadata);
+      // blocking wait, though we could let uploads just run in background
+      upload.waitForCompletion();
+
+      log.debug("uploaded string content to {}", key);
+
+    } catch (Exception e) {
+      log.error("Error uploading string content to: " + key, e);
+      throw new FileUploadException("Error uploading string content to: " + key, e);
     }
-
-    public List<S3ObjectSummary> getObjectSummaries() {
-        List<S3ObjectSummary> objectSummaries = new ArrayList<S3ObjectSummary>();
-        ObjectListing objects = amazonS3.listObjects
-                (s3Bucket, "candidate/");
-        objectSummaries.addAll(objects.getObjectSummaries());
-        while (objects.isTruncated()) {
-            objects = amazonS3.listNextBatchOfObjects(objects);
-            objectSummaries.addAll(objects.getObjectSummaries());
-        }
-        return objectSummaries;
+  }
+
+  public void deleteFile(String key) {
+    amazonS3.deleteObject(s3Bucket, key);
+  }
+
+  public void copyObject(String sourceKey,
+      String destinationKey) {
+    CopyObjectRequest copyObjectRequest = new CopyObjectRequest(s3Bucket, sourceKey, s3Bucket, destinationKey);
+    transferManager.copy(copyObjectRequest);
+  }
+
+  public void copyObject(String sourceBucket,
+      String sourceKey,
+      String destinationBucket,
+      String destinationKey) {
+
+    sourceBucket = sourceBucket != null ? sourceBucket : s3Bucket;
+    destinationBucket = destinationBucket != null ? destinationBucket : s3Bucket;
+    CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceBucket, sourceKey, destinationBucket,
+        destinationKey);
+    transferManager.copy(copyObjectRequest);
+  }
+
+  public void copyBucketContents(String sourceBucket,
+      String sourcePrefix,
+      String destinationBucket,
+      String destinationPrefix)
+      throws AmazonS3Exception {
+    ListObjectsRequest req = new ListObjectsRequest();
+    req.setBucketName(sourceBucket);
+    req.withPrefix(sourcePrefix);
+    List<S3ObjectSummary> contents = amazonS3.listObjects(req).getObjectSummaries();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).